packages/client/src/pages/register.vue

```diff
@@ -1,146 +1,233 @@
-              :rules="[requiredRule]"
-              lazy-rules
-              :label="t('auth.lastName')"
-              outlined
-              type="text"
-              data-cy="surname-field"
-            />
-            <q-input
-              v-model="user.email"
-              :rules="[requiredRule, emailRule]"
-              lazy-rules
-              :label="t('auth.email')"
-              outlined
-              type="email"
-              data-cy="email-field"
-            />
-            <k-password-input
-              v-model="user.password"
-              v-model:show="showPassword"
-              :rules="[requiredRule, validatePasswordRule]"
-              :label="t('auth.password')"
-              outlined
-              lazy-rules
-              autocomplete="password"
-              data-cy="password-field"
-            />
-            <password-strength-bar
-              :password-to-check="user.password"
-              :steps="STRENGTH_BAR_STEPS"
-            />
-            <k-password-input
-              v-model="user.passwordConfirmation"
-              v-model:show="showPassword"
-              :rules="[requiredRule, passwordMatchRule]"
-              :label="t('auth.confirmPassword')"
-              outlined
-              lazy-rules
-              autocomplete="password"
-              data-cy="password-confirmation-field"
-            />
-          </q-card-section>
-
-          <q-card-section>
-            <q-btn
-              class="full-width"
-              color="accent"
-              :label="t('auth.register')"
-              text-color="black-54"
-              type="submit"
-              :loading="isRegistering"
-              data-cy="submit-button"
-            />
-          </q-card-section>
-
-          <template v-if="SOCIAL_LOGIN_ENABLED">
-            <q-separator inset spaced />
-
-            <q-card-section class="column gap-8">
-              <span class="text-black-87">{{ t("common.or") }}</span>
-
-              <social-auth-buttons type="register" />
-            </q-card-section>
-          </template>
-
-          <q-separator inset spaced />
-
-          <q-card-section>
-            <span class="text-dark text-subtitle1">
-              {{ t("common.or") }}&nbsp;
-              <router-link to="login" class="text-dark">
-                {{ t("auth.login") }}
-              </router-link>
-            </span>
-          </q-card-section>
+  <q-page class="justify-center registration-page row">
+    <q-card class="column form-card gap-24 items-stretch no-wrap q-pa-lg">
+      <q-card-section class="column gap-24 no-padding no-wrap">
+        <q-img :src="theme.logo" fit="contain" height="60px" />
+
+        <q-form
+          class="column gap-8 items-stretch justify-center no-padding"
+          greedy
+          @submit="onSubmit"
+        >
+          <q-input
+            v-for="{ field, tooltip, ...inputData } in formData"
+            :key="field"
```

Comment:
I think this one here is fine for now, but please always remember what Callo taught you: premature optimization generally is evil.
Here you are trying to abstract the component of a registration form that can have very particular rules.
Usually, unless you have to replicate the same thing for many pages/components and you find a pattern, it is better and more manageable to just leave the elements coded into the template one by one and bind their own rules there.

For example, I think I noticed an issue here:
- if you set the birth date so that the user is > 18 years old, then no problem.
- But if you set the user to be < 18 years old, then you change the birth date to that the user is > 18 years old, the UI error does not disappear but at the same time the form can be submitted.
![image](https://github.com/ilmercatinodellibro/ilmercatinodellibro/assets/30588533/0f4ae98f-d4fb-4724-afe7-1ee79535d944)